### PR TITLE
cbt: 1.16.2+ support

### LINKED ISSUE #18: Fabric API 0.20 breaks CBT
OS: Mac OS, but bug is likely OS agnostic. 
JVM: 8, 11, 14 (all 64 bit)
Loader version: 0.9.3+build.207
Game version: 1.16.2

Reproducible using just CBT 0.1.2 + Fabric API 0.20.0 build 399

Relevant bit from the log:

(normal log things)
[23:32:18] [main/INFO]: Backend library: LWJGL version 3.2.2 build 10
---- Minecraft Crash Report ----
// Sorry :(

Time: 5/09/20 11:32 PM
Description: Initializing game

java.lang.UnsupportedOperationException
	at com.google.common.collect.ImmutableCollection.add(ImmutableCollection.java:218)
	at net.minecraft.class_3304.handler$zeo000$injectCBTPack(class_3304.java:1029)
	at net.minecraft.class_3304.method_18232(class_3304.java)
	at net.minecraft.class_310.<init>(class_310.java:569)
	at net.minecraft.client.main.Main.main(Main.java:177)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at net.fabricmc.loader.game.MinecraftGameProvider.launch(MinecraftGameProvider.java:192)
	at net.fabricmc.loader.launch.knot.Knot.init(Knot.java:140)
	at net.fabricmc.loader.launch.knot.KnotClient.main(KnotClient.java:26)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.multimc.onesix.OneSixLauncher.launchWithMainClass(OneSixLauncher.java:196)
	at org.multimc.onesix.OneSixLauncher.launch(OneSixLauncher.java:231)
	at org.multimc.EntryPoint.listen(EntryPoint.java:143)
	at org.multimc.EntryPoint.main(EntryPoint.java:34)

Full log: [here](https://paste.ee/p/xIk2p)!

From a brief glance, CBT now seems to be trying to modify a ImmutableCollection, which is very :tiny_potato: in my books.

### PR DESCRIPTION
**Pull request by telans**
Includes fabric 0.2.0+ fix.

Closes #17 
Closes #18 

Co-authored-by: Perry <r3alcl0ud@gmail.com>
Co-authored-by: LambdAurora <aurora42lambda@gmail.com>

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -3,12 +3,12 @@ org.gradle.jvmargs=-Xmx1G
 
 # Fabric Properties
 	# check these on https://fabricmc.net/use
-minecraft_version=1.16.1
-yarn_mappings=1.16.1+build.9
-loader_version=0.8.8+build.202
+	minecraft_version=1.16.3
+	yarn_mappings=1.16.3+build.5
+	loader_version=0.9.3+build.207
+	fabric_version=0.20.2+build.402-1.16
```
Comment (ghost) [inline gradle.properties]: Latest version as of this checking is 0.21.0+build.407-1.16 (fabric_version). Looks good so far though.

File: gradle.properties [hunk 2]
```diff
@@ -3,12 +3,12 @@ org.gradle.jvmargs=-Xmx1G
 
 # Fabric Properties
 	# check these on https://fabricmc.net/use
-minecraft_version=1.16.1
-yarn_mappings=1.16.1+build.9
-loader_version=0.8.8+build.202
+	minecraft_version=1.16.3
+	yarn_mappings=1.16.3+build.5
```
Comment (ghost) [inline gradle.properties]: Latest version of this check is 1.16.3+build.11 (yarn_mappings).

**Review by ghost — APPROVED** [review]
I've tested it as both a runtime dependency of my mod (in the dev environment) as well as with `./gradlew runClient` for your fork of the mod and it runs just fine. I'd say it's good for a PR. We just need @Nuclearfarts to approve the changes.

Jitpack Build Log For Runtime Dependency - https://jitpack.io/com/github/telans/connected-block-textures/1.16-66e0ce0961-1/build.log

**Comment by ghost** [comment]
I guess I should mention, I've only tested for 1.16.3, not 1.16.2.

**Comment by ghost** [comment]
@telans Just wanting to let you know, Fabric API 0.21.0+build.407-1.16 breaks your PR because SpriteAtlasTexture.BLOCK_ATLAS_TEX no longer exists and is replaced by (SpriteAtlasTexture.BLOCK_ATLAS_TEXTURE?)

**Comment by telans** [comment]
> Just wanting to let you know, Fabric API 0.21.0+build.407-1.16 breaks your PR

~~Thanks for the heads up, but it seems to be working with the updated mapping/fabric versions~~

My bad, I guess I hadn't refreshed the sources or something along those lines.

**Comment by ghost** [comment]
The latest pr changes works with the latest version of Fabric API. 🥰

**Comment by Mord3th** [comment]
Do I need to do something other than "gradle build" after checking out the pull request and copying build/libs/connected-block-textures-0.1.3+1.16.jar into my minecraft mods directory?  Minecraft loads with all of the menu text replaced by rectangles, then loops back into its loading animation.  Thanks in advance for any help!

**Comment by ghost** [comment]
Update to the latest version of Fabric API. The loop is from either `SpriteAtlasTexture.BLOCK_ATLAS_TEX` not existing or `SpriteAtlasTexture.BLOCK_ATLAS_TEXTURE` not existing. If you pulled the absolute latest commit, you need to update Fabric API so that `SpriteAtlasTexture.BLOCK_ATLAS_TEXTURE` will exist.

Edit: Just running `./gradlew build` and copying from the libs directory is enough.

**Comment by Mord3th** [comment]
Is there a newer version of Fabric API than the fabric-api-0.21.0+build.407-1.16.jar on CurseForge (September 18th)?  This is the one I’m using.

**Comment by telans** [comment]
Are you sure it's not due to any other mods?

**Comment by redbrain** [comment]
Checked it out, looks good to me!

**Comment by hrmorley34** [comment]
The 'rectangles text' issue is, I believe, an issue with [LambdaBetterGrass](https://github.com/LambdAurora/LambdaBetterGrass) v1.0.0; it's been fixed by [v1.0.1](https://github.com/LambdAurora/LambdaBetterGrass/releases/tag/1.0.1)

File: src/main/java/io/github/nuclearfarts/cbt/config/BaseCTMConfig.java [hunk 3]
```diff
@@ -63,7 +64,9 @@ public BaseCTMConfig(Properties properties, Identifier location, ResourceManager
 		
 		Predicate<Biome> biomeMatcher;
 		if(properties.containsKey("biomes")) {
-			biomeMatcher = Arrays.stream(properties.getProperty("biomes").split(" ")).map(Identifier::new).map(Registry.BIOME::get).collect(Collectors.toCollection(HashSet::new))::contains;
+			@SuppressWarnings("resource")
+			Registry<Biome> biomes = MinecraftClient.getInstance().world.getRegistryManager().get(Registry.BIOME_KEY);
+				biomeMatcher = Arrays.stream(properties.getProperty("biomes").split(" ")).map(Identifier::new).map(biomes::get).collect(Collectors.toCollection(HashSet::new))::contains;
```
Comment (TwilightFlower) [inline src/main/java/io/github/nuclearfarts/cbt/config/BaseCTMConfig.java:69]: irritatered at this indentation

**Review by TwilightFlower — APPROVED** [review]

**Pull request merged**